flaky test: test_ert_qstat_proxy

```
FAILED tests/unit_tests/job_queue/test_ert_qstat_proxy.py::test_many_concurrent_qstat_invocations
                    assert state == CacheState.CACHE_EXISTS
>                   assert (
                        process.returncode == 0
                    ), "Check for race condition if AssertionError"
E                   AssertionError: Check for race condition if AssertionError
E                   assert 1 == 0
E                    +  where 1 = <Popen: returncode: 1 args: [PosixPath('/opt/hostedtoolcache/Python/3.10.12/...>.returncode
```

The error has been reproduced in https://github.com/equinor/ert/actions/runs/6085593995/job/16510174916?pr=6046

```
tests/unit_tests/job_queue/test_ert_qstat_proxy.py::test_many_concurrent_qstat_invocations[59-1000] 011111111111111111111111111111111111111111111111111111111111111111111111111111111110FAILED
...
FAILED tests/unit_tests/job_queue/test_ert_qstat_proxy.py::test_many_concurrent_qstat_invocations[59-1000] - AssertionError: Check for race condition if AssertionError, we failed on invocation 84 out of 800
assert 1 == 0
```

tldr: There is an out-of-order problem in the loop over subprocesses, the 84th process has probably been lucky to fail before process 83 (or process 83 has been unlucky to have been scheduled late) returned with zero exit code (it found the cache file).

This can be mitigated by either increasing the sleeptime (makes the whole test slower), or perhaps accepting a few out-of-order subprocesses. These two measures will only make the test less flaky, not non-flaky.

--- a/tests/unit_tests/job_queue/test_ert_qstat_proxy.py
+++ b/tests/unit_tests/job_queue/test_ert_qstat_proxy.py
@@ -258,15 +258,14 @@ def test_many_concurrent_qstat_invocations(tmpdir, monkeypatch):
     This test will dump something like
     01111111111000000000000000000000000000000000000 to stdout
     where each digit is the return code from the proxy script. Only one sequence of 1's
-    is allowed after the start, later on there should be no failures (that would be
-    errors from race conditions.)
+    is allowed after the start, except for some noise at the end of the sequence of 1's
+    due to out-of-order finalization of the subprocesses started in this test.
     """
     starttime = time.time()
-    invocations = 400
-    sleeptime = 0.03  # seconds. Lower number increase probability of race conditions.
+    sleeptime = 0.02  # seconds between each qstat process started in this test
     # (the mocked qstat backend sleeps for 0.5 seconds to facilitate races)
     cache_timeout = 2  # This is CACHE_TIMEOUT in the shell script
-    assert invocations * sleeptime > cache_timeout  # Ensure race conditions can happen
+    invocations = int(cache_timeout / sleeptime * 2)
 
     monkeypatch.chdir(tmpdir)
     Path("log").mkdir()  # The mocked backend writes to this directory
@@ -292,6 +291,7 @@ def test_many_concurrent_qstat_invocations(tmpdir, monkeypatch):
             CACHE_EXISTS = 2
 
         state = None
+        cache_hits = 0
         for _, process in enumerate(subprocesses):
             process.wait()
             if state is None:
@@ -314,9 +314,13 @@ def test_many_concurrent_qstat_invocations(tmpdir, monkeypatch):
 
             else:
                 assert state == CacheState.CACHE_EXISTS
-                assert (
-                    process.returncode == 0
-                ), "Check for race condition if AssertionError"
+                if process.returncode == 1:
+                    assert cache_hits < 4, (
+                        "Only a few errors due to out-of-order "
+                        "subprocess finalization is allowed"
+                    )
+                if process.returncode == 0:
+                    cache_hits += 1
 
             print(process.returncode, end="")
         print("\n")

